crates/aptos-crypto/src/validatable.rs

```diff
@@ -214,7 +219,29 @@ impl Validate for Ed25519PublicKey {
     type Unvalidated = UnvalidatedEd25519PublicKey;
 
     fn validate(unvalidated: &Self::Unvalidated) -> Result<Self> {
+        // NOTE: We don't need to actually check prime-order subgroup membership because our signature
+        // verification code does it implicitly. The current design is a bit confusing and should be fixed.
+        // to fix it.
         Self::try_from(unvalidated.0.as_ref()).map_err(Into::into)
+
+        // let compressed = curve25519_dalek::edwards::CompressedEdwardsY(unvalidated.0);
+        // let point = compressed
+        //     .decompress()
+        //     .ok_or(CryptoMaterialError::DeserializationError)?;
+        //
+        // // Check if the point lies on a small subgroup. This is required when using curves with a
+        // // small cofactor (e.g., in Ed25519, cofactor = 8).
+        // if point.is_small_order() {
+        //     return Err(anyhow!("{:?}", CryptoMaterialError::SmallSubgroupError));
+        // }
+        //
+        // // NOTE: There is no way to construct an ed25519_dalek::PublicKey directly from the decompressed
+        // // point above. Instead, we have to call from_bytes(), which re-does the work from above.
+        // // Accessing the elliptic-curve point in a deserialized ed25519_dalek::PublicKey is also not
+        // // possible, so we are stuck with doing the same deserialization work twice until the following
+        // // pull request is merged: https://github.com/dalek-cryptography/ed25519-dalek/pull/188
+        // Ok(Ed25519PublicKey::from_bytes_unchecked(&unvalidated.0[..])
+        //     .expect("second deserialization should have succeeded too"))
```

Comment:
intended to leave this section of commented code?

Reply:
Well, I am actually gonna push another commit that removes this `Validatable` implementation for Ed25519, since it is not needed: the validation (i.e., checking the PK is not of low order) is done inside `verify_strict` (i.e., https://github.com/dalek-cryptography/ed25519-dalek/blob/main/src/public.rs#L303).

**Question for you though:** Should I do that in a separate PR, or just push a second commit in this PR?

Reply:
I don't like to leave sections of commented code, that feels like unfinished work but if it'll be removed soon probably fine

Reply:
Yes, it'll be removed in the next PR.